src/lib.rs

```diff
@@ -158,16 +162,27 @@ pub struct BIP44Path {
     pub index: u32,
 }
 
-fn serialize_bip44(path: &BIP44Path) -> Vec<u8> {
+fn serialize_bip44(path: &BIP44Path) -> Result<Vec<u8>, Error> {
     use byteorder::{LittleEndian, WriteBytesExt};
     let mut m = Vec::new();
     let harden = 0x8000_0000;
+
+    if path.purpose > harden
+        || path.coin > harden
+        || path.account > harden
+        || path.change > harden
+        || path.index > harden
+    {
+        return Err(Error::InvalidPath);
+    }
+
     m.write_u32::<LittleEndian>(harden | path.purpose).unwrap();
     m.write_u32::<LittleEndian>(harden | path.coin).unwrap();
     m.write_u32::<LittleEndian>(path.account).unwrap();
     m.write_u32::<LittleEndian>(path.change).unwrap();
     m.write_u32::<LittleEndian>(path.index).unwrap();
```

Comment:
This will lead to confusion.. better not to harden anything here.. 
We should better unify API here with other libs.. let's not "auto-harden" and then leave this without checks..
we can another function that goes from `path: str` to BIP44Path where we do the checks

Reply:
we can bring the code from filecoin-rs here too for API consistency